src/models/propulsion/FGBrushLessDCMotor.cpp

```diff
@@ -142,42 +145,27 @@ void FGBrushLessDCMotor::Calculate(void)
   }
 
   RPM = Thruster->GetRPM();
-  TorqueRequired = abs(((FGPropeller*)Thruster)->GetTorque()); //units [#*ft]
 
-  CurrentRequired = (TorqueRequired * VelocityConstant) / TorqueConstant;
+  TorqueRequired = abs(((FGPropeller*)Thruster)->GetTorque());           //units [#*ft]
+
 
-  //  total current required must include no load current i0
-  CurrentRequired = CurrentRequired + NoLoadCurrent;
- 
   V = MaxVolts * in.ThrottlePos[EngineNumber];
-  
-  //  Delta RPM = (input voltage - currentRequired * coil resistance) * velocity costant
-  DeltaRPM = (V - CurrentRequired * CoilResistance) * VelocityConstant-RPM;
-
-  //  Torque is MaxTorque (stall torque) at 0 RPM and linearly go to 0 at max RPM (MaxVolts*VelocityCostant)
-  //  MaxTorque = MaxCurrent*torqueconstant/velocityconstant*(1-RPM/maxRPM)
-
-  MaxTorque = MaxCurrent / VelocityConstant * TorqueConstant * (1 - RPM / (MaxVolts* VelocityConstant));
-
-  TorqueAvailable = MaxTorque - TorqueRequired;
-  InertiaTorque = (((DeltaRPM/60)*(2.0 * M_PI))/(max(0.00001, in.TotalDeltaT))) * ((FGPropeller*)Thruster)->GetIxx();
- 
-  //  compute acceleration and deceleration phases:
-  //  Acceleration is due to the max delta torque available and is limited to the inertial forces
-
-  if (DeltaRPM >= 0) {
-    TargetTorque = min(InertiaTorque, TorqueAvailable) + TorqueRequired;
-  } else {
-  //  Deceleration is due to braking force given by the ESC and set by parameter deceleration_time 
-    TargetTorque = TorqueRequired - min(abs(InertiaTorque)/(max(DecelerationFactor,0.01)*30),RPM*TorqueConstant/VelocityConstant/VelocityConstant/CoilResistance);
-  }
 
-  EnginePower = ((2 * M_PI) * RPM * TargetTorque) / 60;   //units [#*ft/s]
-  HP = EnginePower /hptowatts*NMtoftpound;                             // units[HP]
+  double Current = (V - RPM / VelocityConstant) / CoilResistance;        // Equation (4) from Drela's document
+
+  if (Current >= 0)
+    TargetTorque = (Current - NoLoadCurrent) / VelocityConstant * TorqueConstant;    // Assuming Kv == Kq
+  else
+    TargetTorque = (Current + NoLoadCurrent) / VelocityConstant * TorqueConstant; // Assuming Kv == Kq
+
+
+  EnginePower = ((2 * M_PI) * max(RPM, 0.0001) * TargetTorque) / 60;              //units [#*ft/s]
```

Comment:
```suggestion
  EnginePower = ((2 * M_PI) * RPM * TargetTorque) / 60;              //units [#*ft/s]
```
I still don't catch why `EnginePower` should not be zero ?

Reply:
This is necessary otherwise EnginePower will always be 0, since at startup RPM=0.